Classes/ExplorerInterface/FLEXExplorerViewController.m

```diff
@@ -808,6 +808,19 @@ - (BOOL)wantsWindowToBecomeKey
     return self.previousKeyWindow != nil;
 }
 
+- (void)toggleToolWithViewControllerProvider:(UIViewController *(^)())future completion:(void(^)())completion
+{
+    void (^presentBlock)() = ^{
+        [self makeKeyAndPresentViewController:future() animated:YES completion:completion];
+    };
+
+    if (self.presentedViewController) {
+        [self resignKeyAndDismissViewControllerAnimated:YES completion:presentBlock];
```

Comment:
Ah, I missed something about this case. I was thinking we would unconditionally dismiss (and not present the new tool) in the case where something was already presented. As-is, I think there are some places where the intention will be to dismiss (e.g. in the `globalsViewControllerDidFinish:` callback) but we'll end up dismissing and re-presenting.

In thinking about this further, I'm not totally sure what the right thing to do is. Do you have any thoughts here?

Reply:
> I think there are some places where the intention will be to dismiss but we'll end up dismissing and re-presenting.

I think I designed it that way on purpose, but I can't remember why. Having just refreshed my memory with what's going on here, I think a sensible default would be to just dismiss the current tool if one is presented and do nothing else.

Reply:
(So, it no longer dismisses then immediately presents something if something is already presented)